cfme/tests/ansible/test_embedded_ansible_tags.py

```diff
@@ -0,0 +1,119 @@
+import fauxfactory
+import pytest
+
+from cfme import test_requirements
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.wait import wait_for
+
+pytestmark = [
+    test_requirements.ansible,
+    pytest.mark.uncollectif(lambda appliance: appliance.version < "5.9",
+                            reason="5.8 is not support tagging via UI")
+]
+
+
+@pytest.fixture(scope='module')
+def enabled_embedded_ansible(appliance):
+    """Enables embedded ansible role"""
+    appliance.server.settings.enable_server_roles("embedded_ansible")
+    appliance.wait_for_embedded_ansible()
+    yield
+    appliance.server.settings.disable_server_roles("embedded_ansible")
+
+
+@pytest.fixture(scope='module')
+def repository(enabled_embedded_ansible, appliance):
+    repositories = appliance.collections.ansible_repositories
+    repository = repositories.create(
+        name=fauxfactory.gen_alpha(),
+        url="https://github.com/quarckster/ansible_playbooks",
+        description=fauxfactory.gen_alpha())
+    view = navigate_to(repository, "Details")
+    if appliance.version < "5.9":
+        refresh = view.browser.refresh
+    else:
+        refresh = view.toolbar.refresh.click
+    wait_for(
+        lambda: view.entities.summary("Properties").get_text_of("Status") == "successful",
+        timeout=60,
+        fail_func=refresh
+    )
+    yield repository
+
+    if repository.exists:
+        repository.delete()
+
+
+@pytest.fixture(scope='module')
+def credential(enabled_embedded_ansible, appliance):
+    credentials_collection = appliance.collections.ansible_credentials
+    credential = credentials_collection.create(
+        "{}_credential_{}".format('Machine', fauxfactory.gen_alpha()),
+        'Machine',
+        username=fauxfactory.gen_alpha(),
+        password=fauxfactory.gen_alpha()
+    )
+    wait_for(
+        func=lambda: credential.exists,
+        message='credential appears on UI',
+        fail_func=appliance.browser.widgetastic.refresh,
+        delay=20,
+        num_sec=240
+    )
+
+    yield credential
+    if credential.exists:
+        credential.delete()
+
+
+@pytest.fixture(scope='module')
+def playbook(appliance, repository):
+    playbooks_collection = appliance.collections.ansible_playbooks
+    return playbooks_collection.all()[0]
+
+
+def check_tag_place(item, tag_place):
+    tag = item.add_tag(details=tag_place)
+    tags = item.get_tags()
+    assert any(
+        object_tags.category.display_name == tag.category.display_name and
+        object_tags.display_name == tag.display_name for object_tags in tags), (
+        "{}: {} not in ({})".format(tag.category.display_name, tag.display_name, str(tags)))
+
+    item.remove_tag(tag=tag, details=tag_place)
```

Comment:
**REQUIRED**
Teardown should be performed via `request.addfinalizer`.

Reply:
hm, I thought this over and better to leave it as part of the test but I will add an additional check if the tag was really deleted